backend/src/packages/auth/auth.service.ts

```diff
@@ -11,6 +15,35 @@ class AuthService {
     this.userService = userService;
   }
 
+  public async verifySignInCredentials(
+    userRequestDto: UserAuthRequestDto,
+  ):Promise<UserSignInResponseDto> {
+    const user = await this.userService.findByEmail(userRequestDto.email);
+    
+    if (!user) {
+      throw new NotFoundError({ 
+        message: 'User not found', 
+        cause: 'No user email in the database' 
+      });
+    }
+
+    return user;
+  }
+
+  public signIn(
+    userRequestDto: UserSignInResponseDto,
+  ):Promise<UserAuthResponseDto> {
+    const mockedUserResponse: UserAuthResponseDto = {
+      id: userRequestDto.id,
+      email: userRequestDto.email,
+      firstName: 'Test',
+      lastName: 'User',
+    };
+    return Promise.resolve({
+      ...mockedUserResponse,
+    });
+  }
```

Comment:
```suggestion
  public async signIn(
    userSignInDto: UserSignInResponseDto,
  ):Promise<UserAuthResponseDto> {
    const { email } = userSignInDto;
  
    const user = await this.userService.findByEmail(email);
    
    if (!user) {
      /** message and cause should be predefined as default values for argument on exception */
      throw new NotFoundError();
    }

    /** firstName|lastName may be mocked up on entity level, because after implementation they may be fulfilled there */
    return user;
  }
```